src/services/contentCssManager.ts

```diff
@@ -0,0 +1,109 @@
+// This is heavily inspired by the http-vue-loader library.
+// See: https://github.com/FranckFreiburger/http-vue-loader/blob/master/src/httpVueLoader.js.
+
+let scopeIdCounter = 0;
+
+const CMS_CONTENT_ATTRIBUTE_NAME = 'data-cms-css';
+
+export default class ContentCssManager {
+  scopeId: number;
+  private styleElement: HTMLStyleElement | null = null;
+
+  constructor(css?: string) {
+    this.scopeId = scopeIdCounter++;
+
+    if (css) {
+      this.add(css);
+    }
+  }
+
+  get scopeAttrs(): { [key: string]: number } {
+    return { [CMS_CONTENT_ATTRIBUTE_NAME]: this.scopeId };
+  }
+
+  get scopeSelector(): string {
+    return `[${CMS_CONTENT_ATTRIBUTE_NAME}="${this.scopeId}"]`;
+  }
+
+  private add(css: string) {
+    const base = this.createTemporaryBaseElement();
+
+    this.styleElement = document.createElement('style');
+    this.styleElement.appendChild(document.createTextNode(css));
+    this.styleElement.setAttribute('data-cms-css-for', this.scopeId.toString());
```

Comment:
It's slightly different - this one is just for ease of debugging instead of actually scoping the css. I should put it into a constant though, so it's more clear.